src/components/cell-tooltip/CellTooltip1DVertical.js

```diff
@@ -30,6 +33,18 @@ export default function CellTooltip1DVertical(props) {
     setX((cellIndex / numCells) * width);
     setParentWidth(width);
     setParentHeight(height);
+    // Compute the y-coordinate of the current gene, considering the other two canvas elements.
+    if (geneIndex !== null && geneIndex !== undefined && numGenes) {
+      const heatmapCanvasElements = Array.from(el.parentNode.querySelectorAll('canvas').values());
+      const geneOffsetY = heatmapCanvasElements
+        .slice(0, -1)
+        .reduce((a, h) => (a + h.getBoundingClientRect().height), 0);
+      const geneHeatmapHeight = heatmapCanvasElements[heatmapCanvasElements.length - 1]
+        .getBoundingClientRect().height;
```

Comment:
If they're not already present, I think it'd be better to add meaningful IDs to the canvas elements and reference those, rather than basing it all on array order. Imagine if a section is added or removed or moved in the heatmap... How can this code make handling that easier?